routing/road_access.cpp

```diff
@@ -1,25 +1,127 @@
 #include "routing/road_access.hpp"
 
+#include <algorithm>
 #include <sstream>
 
+using namespace std;
+
+namespace
+{
+string const kNo = "No";
+string const kPrivate = "Private";
+string const kDestination = "Destination";
+string const kYes = "Yes";
+string const kCount = "Count";
+string const kNames[] = {kNo, kPrivate, kDestination, kYes, kCount};
+
+// *NOTE* Order may be important for users (such as serializers).
+vector<routing::RouterType> const kSupportedRouterTypes = {
+  routing::RouterType::Vehicle,
+  routing::RouterType::Pedestrian,
+  routing::RouterType::Bicycle,
+};
+}  // namespace
+
 namespace routing
 {
-std::string DebugPrint(RoadAccess const & r)
+// RoadAccess --------------------------------------------------------------------------------------
+// static
+vector<RouterType> const & RoadAccess::GetSupportedRouterTypes() { return kSupportedRouterTypes; }
+
+// static
+bool RoadAccess::IsSupportedRouterType(RouterType r)
 {
-  size_t const kMaxIdsToShow = 10;
-  std::ostringstream oss;
-  oss << "RoadAccess: Private roads [";
-  auto const & privateRoads = r.GetPrivateRoads();
-  for (size_t i = 0; i < privateRoads.size(); ++i)
+  return find(kSupportedRouterTypes.begin(), kSupportedRouterTypes.end(), r) !=
+         kSupportedRouterTypes.end();
+}
+
+RoadAccess::Type const RoadAccess::GetType(RouterType routerType, Segment const & segment) const
+{
+  auto const & types = GetTypes(routerType);
+  Segment key(kFakeNumMwmId, segment.GetFeatureId(), segment.GetSegmentIdx(), segment.IsForward());
```

Comment:
There's a todo precisely for that. I'm seeing two possible ways of doing that:
1. Create a struct segment inside RoadAccess and use it as the map key.
2. Refactor routing/segment.hpp.

I'm thinking about 2, but it is out of the scope of this PR.
What do you think?